index.js

```diff
@@ -1,119 +1,157 @@
 const request = require("request");
 const notifier = require("node-notifier");
+const { SKUS, SKUS_AUSTRALIA, SKUS_IPHONE, SKUS_AUSTRALIA_IPHONE, COUNTRIES, PRODUCT_TYPES } = require("./constants");
 
-const { SKUS, SKUS_AUSTRALIA, COUNTRIES } = require("./constants");
-const args = process.argv.slice(2);
+/********** START USER CONFIGURATION ********/
 
+// List of desireable SKUs (for example Ultimate MBP config)
 let favorites = ["MMQX3LL/A", "MKH53LL/A", "MK1A3LL/A", "MK1H3LL/A"];
+
 // Australia uses different SKUs, if passing in AU as the country code,
 // this favorites list will be used instead of the default
 let favoritesAustralia = ["MMQX3X/A","MKH53X/A","MMQW3X/A","MK233X/A"];
 
-const control = "MYD92LL/A";
+// Set this to false for console output only
+const sendNotification = true;
+
+// default store
 let storeNumber = "R172";
-let state = "CO";
-let countryCode = "";
-let skuList = SKUS;
-
-if (args.length > 0) {
-  const passedStore = args[0];
-  const passedCountry = args[1] ?? "US";
-  if (passedStore.charAt(0) === "R") {
-    // All retail store numbers start with R
-    storeNumber = passedStore;
-    state = null;
-  }
-  countryCode = COUNTRIES[passedCountry];
-  if (countryCode === "/au") {
-    skuList = SKUS_AUSTRALIA;
-    favorites = favoritesAustralia;
-  }
-}
 
-const query =
-  Object.keys(skuList)
-    .map((k, i) => `parts.${i}=${encodeURIComponent(k)}`)
-    .join("&") + `&searchNearby=true&store=${storeNumber}`;
+/************ END USER CONFIGURATION ********/
+
+// used for verification of result
+const control = "MYD92LL/A"; // MBP M1
 
-let options = {
-  method: "GET",
-  url: `https://www.apple.com${countryCode}/shop/fulfillment-messages?` + query,
-};
+const args = process.argv.slice(2);
 
-request(options, function (error, response) {
-  if (error) throw new Error(error);
+// All retail store numbers start with R
+const passedStore = args.length && args[0].charAt(0) === "R" ? args[0] : storeNumber;
 
-  const body = JSON.parse(response.body);
-  const storesArray = body.body.content.pickupMessage.stores;
-  let skuCounter = {};
-  let hasStoreSearchError = false;
+const passedCountry = args[1] ? args[1] : "US";
+const passedProductType = args[2] ? args[2] : "MBP";
+const title = passedProductType === PRODUCT_TYPES.MBP ? "Macbook Pro" : "iPhone";
+let skuList = passedProductType === PRODUCT_TYPES.IPHONE ? SKUS_IPHONE : SKUS;
```

Comment:
given that we now have three argument options, i'm wondering if it makes sense to give these proper argument labels, like:

```sh
$ node index.js --store R123 --country US --search IPHONE
```

Reply:
What about prompting via the command line for each argument?

Reply:
Prompting can be annoying vs just passing in args... but perhaps by default it should print example usage ?

Regarding argument labels that does make parsing a little more complex. We could have it accept no labels (as current) to maintain compatibility for anyone who has it in a cron job, as well as labels (but in that case _every_ arg must be labelled).